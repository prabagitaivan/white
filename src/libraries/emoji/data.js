const data = [
  { image: '😁', text: 'grinning face with smiling eyes' },
  { image: '😂', text: 'face with tears of joy' },
  { image: '🙂', text: 'slightly smiling face' },
  { image: '🙃', text: 'upside-down face' },
  { image: '🤔', text: 'thinking face' },
  { image: '🤨', text: 'face with raised eyebrow' },
  { image: '🤯', text: 'exploding head' },
  { image: '😮', text: 'face with open mouth' },
  { image: '😎', text: 'smiling face with with sunglasses' },
  { image: '🤫', text: 'shushing face' },
  { image: '👻', text: 'ghost' },
  { image: '👽', text: 'alien' },
  { image: '👾', text: 'alien monster' },
  { image: '🤖', text: 'robot' },
  { image: '🙈', text: 'see-no-evil monkey' },
  { image: '🙉', text: 'hear-no-evil monkey' },
  { image: '🙊', text: 'speak-no-evil monkey' },
  { image: '👋', text: 'waving hand' },
  { image: '🖖', text: 'vulcan salute' },
  { image: '✌️', text: 'victory hand' },
  { image: '🤟', text: 'love-you gesture' },
  { image: '🤘', text: 'sign of the horns' },
  { image: '👍', text: 'thumbs up' },
  { image: '👊', text: 'oncoming fist' },
  { image: '🙌', text: 'raising hands' },
  { image: '🤝', text: 'handshake' },
  { image: '⛺', text: 'tent' },
  { image: '🚧', text: 'construction' },
  { image: '⛓️', text: 'chains' },
  { image: '💡', text: 'light bulb' },
  { image: '💭', text: 'thought ballon' },
  { image: '🚀', text: 'rocket' },
  { image: '🛸', text: 'flying saucer' },
  { image: '🗺️', text: 'world map' },
  { image: '☁️', text: 'cloud' },
  { image: '☃️', text: 'snowman' },
  { image: '🧊', text: 'ice' },
  { image: '🔥', text: 'fire' },
  { image: '💥', text: 'collision' },
  { image: '✨', text: 'sparkles' },
  { image: '🕸️', text: 'spider web' },
  { image: '🐛', text: 'bug' },
  { image: '🦋', text: 'butterfly' },
  { image: '🐣', text: 'hatching chick' },
  { image: '🔬', text: 'microscope' },
  { image: '🔭', text: 'telescope' },
  { image: '💻', text: 'laptop' },
  { image: '🖥️', text: 'desktop computer' },
  { image: '📱', text: 'mobile phone' },
  { image: '🎨', text: 'artist paletter' },
  { image: '🎁', text: 'wrapped gift' },
  { image: '🍦', text: 'soft ice cream' },
  { image: '🍉', text: 'watermelon' }
]

export default data
