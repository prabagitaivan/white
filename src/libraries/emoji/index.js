import data from './data'

export const getRandomEmoji = () => {
  const limit = data.length - 1
  const index = Math.floor(Math.random() * limit)

  return data[index]
}
