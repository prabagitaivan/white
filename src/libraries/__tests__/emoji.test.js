import data from '../emoji/data'
import { getRandomEmoji } from '../emoji'

describe('libraries emoji', () => {
  describe('data', () => {
    it('return all defined emoji', () => {
      expect(data).toMatchSnapshot()
    })
  })
  describe('main', () => {
    it('get random emoji data', async () => {
      const emoji = getRandomEmoji()
      const found = data.find(
        datum => datum.text === emoji.text && datum.image === emoji.image
      )

      expect(found).not.toBeUndefined()
    })
  })
})
