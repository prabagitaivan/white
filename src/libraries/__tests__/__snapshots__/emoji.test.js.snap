// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`libraries emoji data return all defined emoji 1`] = `
Array [
  Object {
    "image": "😁",
    "text": "grinning face with smiling eyes",
  },
  Object {
    "image": "😂",
    "text": "face with tears of joy",
  },
  Object {
    "image": "🙂",
    "text": "slightly smiling face",
  },
  Object {
    "image": "🙃",
    "text": "upside-down face",
  },
  Object {
    "image": "🤔",
    "text": "thinking face",
  },
  Object {
    "image": "🤨",
    "text": "face with raised eyebrow",
  },
  Object {
    "image": "🤯",
    "text": "exploding head",
  },
  Object {
    "image": "😮",
    "text": "face with open mouth",
  },
  Object {
    "image": "😎",
    "text": "smiling face with with sunglasses",
  },
  Object {
    "image": "🤫",
    "text": "shushing face",
  },
  Object {
    "image": "👻",
    "text": "ghost",
  },
  Object {
    "image": "👽",
    "text": "alien",
  },
  Object {
    "image": "👾",
    "text": "alien monster",
  },
  Object {
    "image": "🤖",
    "text": "robot",
  },
  Object {
    "image": "🙈",
    "text": "see-no-evil monkey",
  },
  Object {
    "image": "🙉",
    "text": "hear-no-evil monkey",
  },
  Object {
    "image": "🙊",
    "text": "speak-no-evil monkey",
  },
  Object {
    "image": "👋",
    "text": "waving hand",
  },
  Object {
    "image": "🖖",
    "text": "vulcan salute",
  },
  Object {
    "image": "✌️",
    "text": "victory hand",
  },
  Object {
    "image": "🤟",
    "text": "love-you gesture",
  },
  Object {
    "image": "🤘",
    "text": "sign of the horns",
  },
  Object {
    "image": "👍",
    "text": "thumbs up",
  },
  Object {
    "image": "👊",
    "text": "oncoming fist",
  },
  Object {
    "image": "🙌",
    "text": "raising hands",
  },
  Object {
    "image": "🤝",
    "text": "handshake",
  },
  Object {
    "image": "⛺",
    "text": "tent",
  },
  Object {
    "image": "🚧",
    "text": "construction",
  },
  Object {
    "image": "⛓️",
    "text": "chains",
  },
  Object {
    "image": "💡",
    "text": "light bulb",
  },
  Object {
    "image": "💭",
    "text": "thought ballon",
  },
  Object {
    "image": "🚀",
    "text": "rocket",
  },
  Object {
    "image": "🛸",
    "text": "flying saucer",
  },
  Object {
    "image": "🗺️",
    "text": "world map",
  },
  Object {
    "image": "☁️",
    "text": "cloud",
  },
  Object {
    "image": "☃️",
    "text": "snowman",
  },
  Object {
    "image": "🧊",
    "text": "ice",
  },
  Object {
    "image": "🔥",
    "text": "fire",
  },
  Object {
    "image": "💥",
    "text": "collision",
  },
  Object {
    "image": "✨",
    "text": "sparkles",
  },
  Object {
    "image": "🕸️",
    "text": "spider web",
  },
  Object {
    "image": "🐛",
    "text": "bug",
  },
  Object {
    "image": "🦋",
    "text": "butterfly",
  },
  Object {
    "image": "🐣",
    "text": "hatching chick",
  },
  Object {
    "image": "🔬",
    "text": "microscope",
  },
  Object {
    "image": "🔭",
    "text": "telescope",
  },
  Object {
    "image": "💻",
    "text": "laptop",
  },
  Object {
    "image": "🖥️",
    "text": "desktop computer",
  },
  Object {
    "image": "📱",
    "text": "mobile phone",
  },
  Object {
    "image": "🎨",
    "text": "artist paletter",
  },
  Object {
    "image": "🎁",
    "text": "wrapped gift",
  },
  Object {
    "image": "🍦",
    "text": "soft ice cream",
  },
  Object {
    "image": "🍉",
    "text": "watermelon",
  },
]
`;
